### PR TITLE
Refactor: Extract kubeclient from cloud provider

### PR DESCRIPTION
**Pull request by MartinForReal**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Refactor: Extract kubeclient from cloud provider
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
the ut coverage is dropped dramatically:
```
coverage/coveralls — Coverage decreased (-2.9%) to 77.335%
```

**Comment by MartinForReal** [comment]
image

File: pkg/blobplugin/main.go [hunk 1]
```diff
@@ -26,45 +26,37 @@ import (
...
 	"sigs.k8s.io/blob-csi-driver/pkg/blob"
+	"sigs.k8s.io/blob-csi-driver/pkg/util"
 
 	"k8s.io/component-base/metrics/legacyregistry"
 	"k8s.io/klog/v2"
 )
 
+var driverOptions blob.DriverOptions
 var (
-	endpoint                               = flag.String("endpoint", "unix://tmp/csi.sock", "CSI endpoint")
```
Comment (andyzhangx) [inline pkg/blobplugin/main.go:35]: it's better putting all flags in one place in main.go, that's easier to managed in long term

**Comment by MartinForReal** [comment]
<html>
<body>

pull-blob-csi-driver-external-e2e-nfs
--


<br class="Apple-interchange-newline">
</body>
</html>

**Comment by MartinForReal** [comment]
I think it is a common practice to define flag only when it is used. It is easier to figure out unused flags. 
The pattern is adopted by k/k https://github.com/search?q=repo%3Akubernetes%2Fkubernetes%20addflag&type=code

File: pkg/blob/blob.go [hunk 2]
```diff
@@ -167,11 +168,29 @@ type DriverOptions struct {
 	SasTokenExpirationMinutes              int
 }
 
+func (option *DriverOptions) AddFlags() {
```
Comment (andyzhangx) [inline pkg/blob/blob.go:171]: if you could add ut for this func, it should improve the test coverage, current num is below 80%

Comment (MartinForReal) [reply pkg/blob/blob.go:171]: please refer to TestDriverOptions_AddFlags

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**